src/block/decompress_safe.rs

```diff
@@ -293,26 +261,23 @@ fn duplicate_slice(
         if did_overflow_1 {
             return Err(DecompressError::OffsetOutOfBounds);
         }
-        let output_end = output.pos() + match_length;
-        if output_end + (16 - 1) <= output.capacity() {
-            for i in (start..start + match_length).step_by(16) {
-                output.output.copy_within(i..i + 16, output.pos());
-                output.pos += 16;
+        match match_length {
+            0..=32 if output.pos() + 32 <= output.capacity() => {
+                output.extend_from_within_wild(start, 32, match_length)
+            }
+            33..=64 if output.pos() + 64 <= output.capacity() => {
+                output.extend_from_within_wild(start, 64, match_length)
```

Comment:
previously we had n 16bytes block copies, now we switch to a single 32 or 64 byte copy or one large one.
32 bytes or 64 bytes should be fine, but the benchmarks may not reflect the last case, I'm not sure

Reply:
Good point, I don't think they do. I think this is better most of the time (except maybe between 65..=127 bytes).

Some thoughts:
* I couldn't keep the existing version as set_pos is inherently unsafe in a Sink abstraction (it exposes uninitiated data temporarily until set_len is called to truncate). I tried to modify it in different ways to prevent that but they all caused regressions.
* This new version is  significantly faster for the Json dataset (which is common and realistic). The other datasets are mostly hitting the hot-path so this doesn't affect them.
* At 64b+ copies the memcpy call overhead is increasingly small and memcpy surely takes over after a point (I'm guessing around 128B, maybe less).